lib/rucio/tests/test_multi_vo.py

```diff
@@ -115,6 +115,37 @@ def test_update_vo(self):
                 vo_update_success = True
         assert_true(vo_update_success)
 
+    def test_super_root_permissions(self):
+        """ MULTI VO (CORE): Test super_root cannot access root/user functions """
+        rse_str = ''.join(choice(ascii_uppercase) for x in range(10))
+        rse_name = 'MOCK_%s' % rse_str
+        scope_uuid = str(generate_uuid()).lower()[:16]
+        scope = 'mock_%s' % scope_uuid
+
+        # Test an account name that exists at one VO cannot be found if paired with a different VO
+        with assert_raises(AccountNotFound):
+            add_scope(scope, 'super_root', 'super_root', **self.vo)
+
+        # Test super_root@def with functions at vo='def'
+        with assert_raises(AccessDenied):
+            add_rse(rse_name, 'super_root', vo='def')
+        with assert_raises(AccessDenied):
+            add_scope(scope, 'root', 'super_root', vo='def')
+        add_scope(scope, 'super_root', 'super_root', vo='def')
+        assert_in(scope, [s for s in list_scopes(filter={}, vo='def')])
+
+        # Test the permissions of a user account with name super_root
```

Comment:
Would it be sensible to ban new users from using the name 'super_root'?

Reply:
Given that super_root permissions (currently) rely on both the name 'super_root' and vo='def' there shouldn't ever be an actual issue caused by naming a new account as such (I wrote the tests to prove this to myself, then left them in in case at some point down the line the permissions were changed to validate on say the external representation of the issuer).
It might be worth doing purely from a clarity standpoint (e.g. so VOs don't think they should/could add a 'super_root' account to access the super_root functions - but even then they shouldn't really be aware of those functions in the first place). But on the whole I don't think it would matter too much either way, unless I've overlooked something.
Although having said all that I've just now thought I'm not sure what would happen if you named a user account 'super_root' in single VO mode (since the VO for that would be 'def'). If that causes some weirdness it might be easiest to limit the name of new accounts as you say.

Reply:
So I've had a bit of an experiment, if you're in single_vo mode you **can** create an account named `super_root` that will then pass all the permissions for the `add_vo` etc. functions (the fact that `mutli_vo` is not set in `rucio.cfg` doesn't seem to affect anything - it only seems to be used in a few of the daemons and the tests). However, my understanding was these functions would only be accessible from the API/core layer (#17) so this might not actually be an issue? But actually there is [implementation for `add_vo` and `list_vo` at the REST level](https://github.com/stfc/rucio/blob/feature-2635-multiVO_functionality/lib/rucio/web/rest/webpy/v1/vo.py) (from Andew's time?) so maybe this isn't the case (or else this should be removed)?
Simply restricting the use of the name `super_root` might be the most expedient thing to do, but it might be possible that accounts with this name could already exist in single_vo instances, in which case simply restricting new accounts may not be enough.
I'd be keen to hear what you think about this (if you have time) but it seems like it might be something to run past Martin (especially in light of the vulnerability of the old rse_expression stuff - if this is going to be a similar issue it'd be best to identify it before working on it further).

Reply:
Yes, I was going to bring up the single_vo super_root issue. I think we could add a check to the permissions that confirms the config has `multi_vo=True` for `add_vo` etc, unless that's harder to get hold of at the core layer.
I wasn't aware of the `add_vo`/`list_vo` functions in the REST layer, let's ask Martin about that and also about whether we should restrict use of `super_root` as a username.

Reply:
Following discussion with Martin, decided to:
 - Prevent creating new accounts with the name `super_root`
 - Use the `mutli_vo = True` flag to lock out the `super_root` functions while in single VO mode
 - Add the other `super_root` functions to the REST layer but by default have the configuration for them commented out to prevent accidental access